src/model/SpecificationExtension.ts

```diff
@@ -6,14 +6,14 @@
 
 //  Specification Extensions
 //   ^x-
-export interface ISpecificationExtension {
-    // Cannot constraint to "^x-" but can filter them later to access to them
-    [extensionName: string]: any;
-}
+export type IExtensionName = `x-${string}`;
```

Comment:
With newer versions of typescript you can actually strongly type a string prefix, which is nice